### PR TITLE
[12] [MIG] Account Analytic No Lines

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
Migration of account_analytic_no_lines from 11.0 to 12.0.
11.0 migration : #190 
#189

### CODE REVIEW

File: account_analytic_no_lines/models/account_analytic_account.py
Comment (espo-tony) [inline account_analytic_no_lines/models/account_analytic_account.py:14]: For sake of inheritance simplicity it could be better to put the domain creation in a separate method and invoke it here.

File: account_analytic_no_lines/__manifest__.py
Comment (nikul-serpentcs) [inline account_analytic_no_lines/__manifest__.py]: @ThomasBinsfeld Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10)

File: account_analytic_no_lines/views/account_move_line.xml
Comment (nikul-serpentcs) [inline account_analytic_no_lines/views/account_move_line.xml:14]: Directly use `field`  here

**Review by nikul-serpentcs — COMMENTED** [review]
Please squash commits, Improve Code

File: account_analytic_no_lines/views/account_move_line.xml
Comment (ThomasBinsfeld) [reply account_analytic_no_lines/views/account_move_line.xml:14]: This xpath doesn't target the field but the group just before (which has no name).

File: account_analytic_no_lines/models/account_analytic_account.py
Comment (ThomasBinsfeld) [reply account_analytic_no_lines/models/account_analytic_account.py:14]: @espo-tony I won't refactor the code in this migration PR. We can do that in a refactoring PR after the merge.

File: account_analytic_no_lines/models/account_move_line.py
Comment (florian-dacosta) [inline account_analytic_no_lines/models/account_move_line.py:23]: @ThomasBinsfeld 
The method is never actually called here right?
it should be `self = self.filtered(self._must_create_analytic_line())`
And `_must_create_analytic_line` should have an api.multi decorator and manage multiples account move line. (and remove the `move_line` parameter)

That said, I wonder, why not make it simpler and just totally by pass `create_analytic_lines`?
I am not really confortable with the "expense policy" of the products, but I wonder, it would not make sense for someone who need analytic lines for products with special expense policies to install this module right?

Problem is also present in version 10 and 11.

**Review by florian-dacosta — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
